assembly/src/parsers/ast/mod.rs

```diff
@@ -0,0 +1,563 @@
+                            )),
+                        },
+                    }?;
+                    tokens.advance();
+
+                    nodes.push(Node::While(loop_body))
+                }
+                // ------------------------------------------------------------------------------------
+                Token::REPEAT => {
+                    // record start of the repeat block and consume the 'repeat' token
+                    let repeat_start = tokens.pos();
+                    tokens.advance();
+
+                    // read the loop body
+                    let loop_body = self.parse_body_token(tokens, program)?;
+
+                    // consume the `end` token
+                    match tokens.read() {
+                        None => Err(AssemblyError::unmatched_repeat(
+                            tokens.read_at(repeat_start).expect("no repeat token"),
+                        )),
+                        Some(token) => match token.parts()[0] {
+                            Token::END => token.validate_end(),
+                            Token::ELSE => Err(AssemblyError::dangling_else(token)),
+                            _ => Err(AssemblyError::unmatched_repeat(
+                                tokens.read_at(repeat_start).expect("no repeat token"),
+                            )),
+                        },
+                    }?;
+                    tokens.advance();
+
+                    nodes.push(Node::Repeat(repeat_start, loop_body))
+                }
+                Token::EXEC => {
+                    let label = token.parse_exec()?;
+                    tokens.advance();
+                    if label.contains(MODULE_PATH_DELIM) {
+                        let full_proc_name = self.get_full_imported_proc_name(label);
+                        nodes.push(Node::Instruction(Instruction::ExecImported(full_proc_name)))
+                    } else {
+                        let index = program.procedures.get(&label).unwrap().index;
+                        nodes.push(Node::Instruction(Instruction::ExecLocal(index)))
+                    }
+                }
+                Token::CALL => {
+                    let label = token.parse_call()?;
+                    tokens.advance();
+                    if label.contains(MODULE_PATH_DELIM) {
+                        let full_proc_name = self.get_full_imported_proc_name(label);
+                        nodes.push(Node::Instruction(Instruction::CallImported(full_proc_name)))
+                    } else {
+                        let index = program.procedures.get(&label).unwrap().index;
+                        nodes.push(Node::Instruction(Instruction::CallLocal(index)))
+                    }
+                }
+                Token::END => {
+                    token.validate_end()?;
+                }
+                Token::USE | Token::EXPORT | Token::PROC | Token::BEGIN => {}
```

Comment:
This should never happen, right? If so, rather than just doing nothing, I'd put `unreacheable!` call in here.

Reply:
fixed